crates/sui-rosetta/src/operations.rs

```diff
@@ -558,6 +558,83 @@ impl Operations {
         };
         balance_change.chain(gas)
     }
+
+    fn is_single_gascoin_transfer(tx: SuiTransactionBlockKind) -> bool {
+        match tx {
+            SuiTransactionBlockKind::ProgrammableTransaction(pt) => {
+                let SuiProgrammableTransactionBlock {
+                    inputs: _,
+                    commands,
+                } = &pt;
+                return commands
+                    .into_iter()
+                    .find(|command| match command {
+                        SuiCommand::TransferObjects(objs, _) => {
+                            objs.len() > 0 && objs[0] == SuiArgument::GasCoin
+                        }
+                        _ => false,
+                    })
+                    .is_some();
+            }
+            _ => {}
+        }
+        false
+    }
+
+    fn process_single_gascoin_transfer(
+        coin_change_operations: &mut impl Iterator<Item = crate::operations::Operation>,
+        tx: SuiTransactionBlockKind,
+        prev_gas_owner: SuiAddress,
+        new_gas_owner: SuiAddress,
+        gas_used: i128,
+    ) -> Vec<Operation> {
+        let mut operations = vec![];
+        if Self::is_single_gascoin_transfer(tx) {
+            coin_change_operations.into_iter().for_each(|operation| {
+                match operation.type_ {
+                    OperationType::Gas => {
+                        // change gas account back to the previous owner as it is the one
+                        // who paid for the txn (this is the format Rosetta wants to process)
+                        operations.push(Operation::gas(prev_gas_owner, gas_used))
+                    }
+                    OperationType::SuiBalanceChange => {
```

Comment:
What happens if multiple balance changes exist there for sender or recipient? I think this block might double-count gas.

Reply:
balance_changes are derived from `effects` field and even though there were multiple transfers for sender/recipient, there is one entry per address so I think there is no risk of double-count here.

Reply:
Could there be the case though, that the sender also receives an amount equal to the gas-coin transferred, so that there is no balance-change for them? In that case this operation mapping would fail, correct? See also above comment about explicit handling versus mapping.